File: tests/rspec/pages/addons/rte_settings.rb
class RTESettings < ControlPanelPage
  set_url_matcher /addons\/settings\/rte/

  element :title, '.section-header__title'
  elements :headings, 'form h1'

  # Settings View
  element :rte_enabled, 'input[name=rte_enabled]', :visible => false
  element :rte_enabled_toggle, 'a[data-toggle-for=rte_enabled]'
  elements :default_tool_set, 'input[name="rte_default_toolset_id"]'
  element :save_settings_button, 'div.form-btns.form-btns-top input.btn[type="submit"]'

  element :create_new_button, 'div.tbl-ctrls form fieldset.tbl-search a.btn.action'
  elements :tool_sets, 'div.tbl-ctrls form div.tbl-wrap table tr'
  element :tool_set_name_header, 'div.tbl-ctrls form div.tbl-wrap table tr th:first-child'
  element :status_header, 'div.tbl-ctrls form div.tbl-wrap table tr th:nth-child(2)'
  element :manage_header, 'div.tbl-ctrls form div.tbl-wrap table tr th:nth-child(3)'
  element :checkbox_header, 'div.tbl-ctrls form div.tbl-wrap table tr th:nth-child(4)'

  elements :tool_set_names, 'div.tbl-ctrls form div.tbl-wrap table tr td:first-child'
  elements :statuses, 'div.tbl-ctrls form div.tbl-wrap table tr td:nth-child(2)'

  element :bulk_action, 'form fieldset.tbl-bulk-act select[name="bulk_action"]'
  element :action_submit_button, 'form fieldset.tbl-bulk-act input.submit'

  # Tool Set View
  element :tool_set_name, 'input[name="toolset_name"]'
  elements :choose_tools, 'div[data-input-value="tools"] input[type="checkbox"]'
  element :tool_set_save_and_close_button, 'div.form-btns.form-btns-top button[value="save_and_close"]'
  element :tool_set_save_button, 'div.form-btns.form-btns-top button[value="save"]'

  def load
    self.open_dev_menu
    click_link 'Add-Ons'
    self.find('tr', :text => 'Rich Text Editor').find('ul.toolbar li.settings a').click
  end
end

File: tests/cypress/cypress/elements/pages/addons/RteSettings.js
import ControlPanel from '../ControlPanel'

class RteSettings extends ControlPanel {
    constructor() {
        super()
        this.url = 'admin.php?/cp/addons/settings/rte';

        this.selectors = Object.assign(this.selectors, {
            "title": '.main-nav__title',
            "headings": 'form h1',

            // Settings View
            "default_tool_set": 'input[type!=hidden][name="rte_default_toolset"]',
            "save_settings_button": '[value="Save Settings"]',

            "create_new_button": 'a:contains("Create New")',
            "tool_sets": '.table-responsive table tr',
            "tool_set_name_header": '.table-responsive table tr th:first-child',
            "manage_header": '.table-responsive table tr th:nth-child(2)',
            "checkbox_header": '.table-responsive table tr th:nth-child(3)',

            "tool_set_names": '.table-responsive table tr td:first-child a',

            // Tool Set View
            "tool_set_name": 'input[type!=hidden][name="toolset_name"]',
            "choose_tools": 'div[data-input-value="tools"] .field-inputs input[type="checkbox"]',
            "tool_set_save_and_close_button": '.ee-main__content .form-btns-top button[value="save_and_close"]',
            "tool_set_save_button": '.ee-main__content .form-btns-top .button.button--primary'
        })
    }

    confirmSettings() {
        //this.get('breadcrumb').contains('Add-Ons')
        this.get('breadcrumb').contains('Rich Text Editor')

        cy.get('h1').contains('Rich Text Editor')

        this.get('default_tool_set').should('exist')
        this.get('save_settings_button').should('exist')
        this.get('create_new_button').should('exist')
        this.get('tool_sets').should('exist')

        this.get('tool_set_name').should('not.exist')
        this.get('choose_tools').should('not.exist')
    }

    confirmToolset() {
        this.get('breadcrumb').contains('Add-Ons')
        this.get('breadcrumb').contains('Rich Text Editor Configuration')
        this.get('breadcrumb').contains('RTE Tool Set')

        this.get('headings').eq(0).contains('RTE Tool Set')

        this.get('default_tool_set').should('exist')
        this.get('create_new_button').should('exist')
        this.get('tool_sets').should('exist')

        this.get('tool_set_name').should('not.exist')
        this.get('choose_tools').should('not.exist')
        this.get('tool_set_save_and_close_button').should('not.exist')
    }

}

export default RteSettings;
